src/envoy/http/backend_auth/filter.cc

```diff
@@ -53,20 +53,10 @@ const Envoy::Http::LowerCaseString kXForwardedAuthorization{
 FilterHeadersStatus Filter::decodeHeaders(RequestHeaderMap& headers, bool) {
   // Make sure route is calculated
   auto route = decoder_callbacks_->route();
-  if (route == nullptr || route->routeEntry() == nullptr) {
-    config_->stats().denied_by_no_route_.inc();
 
-    rejectRequest(
-        Envoy::Http::Code::NotFound,
-        absl::StrCat("Request `", utils::readHeaderEntry(headers.Method()), " ",
-                     utils::readHeaderEntry(headers.Path()),
-                     "` is not defined by this API."),
-        utils::generateRcDetails(utils::kRcDetailFilterBackendAuth,
-                                 utils::kRcDetailErrorTypeUndefinedRequest));
-    decoder_callbacks_->streamInfo().setResponseFlag(
-        Envoy::StreamInfo::ResponseFlag::NoRouteFound);
-    return FilterHeadersStatus::StopIteration;
-  }
+  // This shouldn't happen as the fallback wildcard route match should catch all
+  // the undefined requests.
+  ASSERT(route != nullptr && route->routeEntry() != nullptr);
```

Comment:
I'm confused now. Won't `route` be `nullptr` and we should return `FilterHeadersStatus::Continue`? This is what you had in your first commit, but it seems like you changed it to `ASSERT`.

And actually I see a few e2e tests failed with 503. A quick look at the logs show crashes, but I cannot find the full stacktrace. Assuming it's this `ASSERT`

![image](https://user-images.githubusercontent.com/11142171/103043477-faa5d300-454a-11eb-8d40-b8ed34160ea9.png)

Reply:
Here actually the problem is routeEntry is null for the catch-all route match as we don't put any per-route-config there.